kt/godot-runtime/src/main/kotlin/godot/core/GarbageCollector.kt

```diff
@@ -81,37 +106,42 @@ object GarbageCollector {
         staticInstances.add(instance)
     }
 
-    fun getObjectInstance(ptr: VoidPtr): KtObject? {
+    fun getObjectInstance(ptr: VoidPtr, id: Long): KtObject? {
         val ktObject = wrappedMap[ptr]
         if (ktObject != null) {
-            if (MemoryBridge.checkInstance(ptr, ktObject.godotInstanceId)) {
+            if (id == ktObject.__id) {
                 return ktObject
-            } else {
-                return null
             }
-        } else {
-            return null
         }
+        return null
     }
 
-    fun getRefInstance(ptr: VoidPtr): KtObject? {
-        return refWrappedMap[ptr]?.get()
+    fun getRefInstance(index: Int): KtObject? {
```

Comment:
That actually something I was wondering. Int (32bits) can go up to around 2 billion, so yeah it's possible to reach that limit.
I don't think widening it to Long (64 bits) should have any measurable cost. And that would make it impossible to reach the limit this way.

Reply:
Well after checking it's actually not possible.
Godot holds a list of all Objects currently alive and it can't be bigger than 24 bits, so we will never go above the limit of a int32